### DISCUSSION TITLE
Connecting to SQLite Database with Python

### DISCUSSION THREAD

**Discussion by Hayessc2** [f1db/f1db#30] [Q&A] [answered]
Hello, I'm new to using Python and navigating the GitHub ecosystem. Could someone please provide me with instructions/examples on how to connect to the SQLite database? I'm interested in querying the data using Python in order to do some analysis. Thank you!

**Comment by msamedozmen** [comment]
Hello, you need to take a look at sqlite3 module documentation to understand concepts. But it is easy to understand and apply in python language.  You can watch the sqlite3 course which was made by freeCodeCamp here is the [link](https://www.youtube.com/watch?v=byHcYRpMgI4). It will help u to grasp the fundamentals of sqlite3. 

Once you understand how it works. You need to dig into pandas, numpy, seaborn, and matplotlib modules for data cleaning and data visualization. I usually use PostgreSQL to store my data. After that, I convert my database tables to data frames. After, that I used Pandas and numpy for data cleaning. Once I clean the data, I'll use matplotlib and seaborn for data visualization. 


If you want to build some dashboards. You can learn and use streamlit module. 

Basically your steps should be like:

-Understanding the fundamentals of Python.
-Learn to use the sqlite3 module.
-Dig into Pandas, Numpy, Matplotlib, and Seaborn 



Example of connection sqlite3:
import sqlite3
conn = sqlite3.connect("databaname.db")
cursor = conn.cursor()
cursor.execute("""CREATE TABLE person(
          first_name = TEXT,
          lastname= TEXT,
          age = INTEGER
           )
"""
cursor.commit()

cursor.execute("INSERT INTO person VALUES('NAME' , 'LASTNAME' , 17)")

cursor.commit()

**Comment by marceloverdijk** [comment, marked as answer]
Hope this helps:

```
import sqlite3

# Replace with the actual path to the database file you downloaded
db_path = 'f1db.db'

# Connect to the SQLite database
conn = sqlite3.connect(db_path)

# Create a cursor object
cur = conn.cursor()

# Write your SQL query. This is an example query; you should replace it with your actual query.
query = "SELECT * FROM driver LIMIT 10;"

try:
    # Execute the query
    cur.execute(query)

    # Fetch and print the query results
    for row in cur.fetchall():
        print(row)
except Exception as e:
    print(f"An error occurred: {e}")
finally:
    # Close the cursor and connection to the database
    cur.close()
    conn.close()
```
